frontend/src/metabase/forms/components/FormProvider/FormProvider.tsx

```diff
@@ -17,15 +19,23 @@ export function FormProvider<T extends FormikValues, C = unknown>({
   onSubmit,
   ...props
 }: FormProviderProps<T, C>): JSX.Element {
-  const { state, handleSubmit } = useFormSubmit({ onSubmit });
+  const { state, handleSubmit, setState } = useFormSubmit({ onSubmit });
   const { initialErrors, handleValidate } = useFormValidation<T, C>({
     initialValues,
     validationSchema,
     validationContext,
   });
+  const value = useMemo(
+    () => ({
+      ...state,
+      setStatus: (status: FormStatus) =>
+        setState(state => ({ ...state, status })),
+    }),
+    [state, setState],
+  );
```

Comment:
I don't hate this, but I wonder if you could avoid it by having different forms for different strategies?

Reply:
I've scheduled an exploration of this idea for Milestone 2. I haven't tried this approach yet because I think it would probably mean putting the strategy radio button outside the form (though I might be wrong).